rpm/s3rpm.py

```diff
@@ -0,0 +1,195 @@
+from __future__ import print_function
+from pyrpm.yum import YumPackage
+from pyrpm.tools.createrepo import YumRepository
+import boto3
+import os
+import botocore
+import gnupg
+import json
+import shutil
+def lambda_handler(event, context):
+    s3 = boto3.client('s3')
+
+    bucket = event['Records'][0]['s3']['bucket']['name']
+    key = event['Records'][0]['s3']['object']['key']
+    repo = YumRepository('/tmp/repo/') # set repository
+    prefix = '/'.join(key.split('/')[0:-1])+'/'
+
+    if os.environ['REPO_DIR'].endswith('/'):
+        os.environ['REPO_DIR'] = os.environ['REPO_DIR'][:-1]
+    if os.environ['REPO_DIR'].startswith('/'):
+        os.environ['REPO_DIR'] = os.environ['REPO_DIR'][1:]
```

Comment:
I think `os.environ[X]` should be treated as a readonly global variable even though python allows you to change it. Updates like this can introduce some weird and unexpected side effects.
```python
repoDir  = os.environ['REPO_DIR'].strip('/') # is good enough :)
```